config/config.go

```diff
@@ -119,53 +127,49 @@ type Nodepool struct {
-	AWSSecurityGroupNames               []string `json:"awsSecurityGroupNames" yaml:"awsSecurityGroupNames"`
-	AWSSecurityGroups                   []string `json:"awsSecurityGroups" yaml:"awsSecurityGroups"`
-	AWSSubnetID                         string   `json:"awsSubnetID" yaml:"awsSubnetID"`
-	AWSSubnets                          []string `json:"awsSubnets" yaml:"awsSubnets"`
-	AWSVpcID                            string   `json:"awsVpcID" yaml:"awsVpcID"`
-	AWSZoneLetter                       string   `json:"awsZoneLetter" yaml:"awsZoneLetter"`
-	AzureClientID                       string   `json:"azureClientID" yaml:"azureClientID"`
-	AzureClientSecret                   string   `json:"azureClientSecret" yaml:"azureClientSecret"`
-	AzureSubscriptionID                 string   `json:"azureSubscriptionID" yaml:"azureSubscriptionID"`
-	CloudCredentialName                 string   `json:"cloudCredentialName" yaml:"cloudCredentialName"`
-	DefaultClusterRoleForProjectMembers string   `json:"defaultClusterRoleForProjectMembers" yaml:"defaultClusterRoleForProjectMembers"`
-	EnableNetworkPolicy                 bool     `json:"enableNetworkPolicy" yaml:"enableNetworkPolicy"`
-	GKENetwork                          string   `json:"gkeNetwork" yaml:"gkeNetwork"`
-	GKEProjectID                        string   `json:"gkeProjectID" yaml:"gkeProjectID"`
-	GKESubnetwork                       string   `json:"gkeSubnetwork" yaml:"gkeSubnetwork"`
-	HostnamePrefix                      string   `json:"hostnamePrefix" yaml:"hostnamePrefix"`
-	LinodeImage                         string   `json:"linodeImage" yaml:"linodeImage"`
-	LinodeRootPass                      string   `json:"linodeRootPass" yaml:"linodeRootPass"`
-	LinodeToken                         string   `json:"linodeToken" yaml:"linodeToken"`
-	MachineConfigName                   string   `json:"machineConfigName" yaml:"machineConfigName"`
-	Module                              string   `json:"module" yaml:"module"`
-	NetworkPlugin                       string   `json:"networkPlugin" yaml:"networkPlugin"`
-	NodeTemplateName                    string   `json:"nodeTemplateName" yaml:"nodeTemplateName"`
-	OSDiskSizeGB                        int64    `json:"osDiskSizeGB" yaml:"osDiskSizeGB"`
-	PrivateAccess                       bool     `json:"privateAccess" yaml:"privateAccess"`
-	ProviderVersion                     string   `json:"providerVersion" yaml:"providerVersion"`
-	PublicAccess                        bool     `json:"publicAccess" yaml:"publicAccess"`
-	Region                              string   `json:"region" yaml:"region"`
-	ResourceGroup                       string   `json:"resourceGroup" yaml:"resourceGroup"`
-	ResourceLocation                    string   `json:"resourceLocation" yaml:"resourceLocation"`
-	VMSize                              string   `json:"vmSize" yaml:"vmSize"`
+	AWSConfig                           AWSConfig               `json:"awsConfig" yaml:"awsConfig"`
```

Comment:
hey, can you update all these vars to `omitempty` so that they aren't required?

Reply:
Resolved